class.jetpack.php

```diff
@@ -6967,4 +6967,23 @@ public static function get_file_url_for_environment( $min_path, $non_min_path )
 
 		return plugins_url( $path, JETPACK__PLUGIN_FILE );
 	}
+
+	/**
+	 * Checks for whether Jetpack Rewind is enabled.
+	 * @return bool|int|mixed
+	 */
+	public static function is_rewind_enabled() {
+		if ( ! Jetpack::is_active() ) {
+			return false;
+		}
+
+		$rewind_enabled = get_transient( 'jetpack_rewind_enabled' );
+		if ( false === $rewind_enabled ) {
+			jetpack_require_lib( 'core-api/class.jetpack-core-api-site-endpoints.php' );
+			$features = Jetpack_Core_API_Site_Endpoint::get_features();
+			$rewind_enabled = ! empty( $features['rewind'] ) ? 1 : 0;
+			set_transient( 'jetpack_rewind_enabled', $rewind_enabled, 10 * MINUTE_IN_SECONDS );
+		}
+		return $rewind_enabled;
```

Comment:
any thoughts on an early-abort-over-nested pattern, even though we can't avoid _some_ nesting here?

```php
if ( false !== $rewind_enabled ) {
	return $rewind_enabled;
}

jetpack_require_lib( … )
…
set_transient( … )

return $rewind_enabled;
```

anyway no big deal just noticed this and how the "main code" was nested

Reply:
Eh, I'm not huge on it either way, this is just how I'd always structured transient checks in the past.  I'm not emotionally invested either direction.